eitprocessing/datahandling/mixins/equality.py

```diff
@@ -14,19 +14,24 @@ class Equivalence:
     """Mixin class that adds an equality and equivalence check."""
 
     # inspired by: https://stackoverflow.com/a/51743960/5170442
-    def __eq__(self, other: Self) -> bool:
+    def __eq__(self, other: object) -> bool:
         if self is other:
             return True
 
         if type(self) is not type(other):
             return False
 
         if is_dataclass(self):
-            attrs_self = Equivalence._return_equality_attributes(vars(self))
-            attrs_other = Equivalence._return_equality_attributes(vars(other))
-            if set(attrs_self.keys()) != set(attrs_other.keys()):
+            field_names = {field.name for field in fields(self)}
+            if set(vars(self).keys()) != field_names or set(vars(other).keys()) != field_names:
```

Comment:
can the first statement ever be False?

Reply:
Yes. This was the case when `_check_equivalence` was still set in `__post_init__`. Attributes that are not defined in the dataclass but are added later show up in `vars(self).keys()`, but not in `fields(self)`.